prophet/handlers/pricepred/client.go

```diff
@@ -137,42 +104,61 @@ type BacktestingResponse struct {
 }
 
 func (c *client) Backtesting(ctx context.Context, req BacktestingRequest) (BacktestingResponse, error) {
-	reqCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	return post[BacktestingRequest, BacktestingResponse](ctx, c.c, 10*time.Second, req, c.backtestingURL)
+}
+
+type VerifyRequest struct {
+	SolverRequest     PredictRequest `json:"solverRequest"`
+	SolverReceipt     ResponseSolverReceipt  `json:"solverReceipt"`
+	VerificationRatio float64        `json:"verificationRatio"`
+}
+
+type VerifyResponse struct {
+	CountItems int  `json:"countItems"`
+	IsVerified bool `json:"isVerified"`
+}
+
+func (c *client) Verify(ctx context.Context, req VerifyRequest) (VerifyResponse, error) {
+	return post[VerifyRequest, VerifyResponse](ctx, c.c, 5*time.Second, req, c.verifyURL)
+}
+
+func post[Req, Resp any](ctx context.Context, client *http.Client, timeout time.Duration, req Req, URL string) (Resp, error) {
```

Comment:
i would change the signature to be this:

```suggestion
func (c *client) post(ctx context.Context, req, res any, URL string) (error) {
```

this way, the struct for Resp is allocated by the caller and has more chances of being allocated on the stack instead of the heap, plus we avoid generics, that to me means a simpler code

we also can use `(c *client)` instead of passing the http.Client, while for the timeout...I don't know, I'm fine either by just passing the ctx or by setting the timeout inside `post()`